File: sast-go/cases/completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_001_T/conditional_switch_001_T.go
package main
import "os/exec"


// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->流程控制语句->条件语句
// scene introduction = switch
// level = 2
// bind_url = completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_001_T/conditional_switch_001_T
// evaluation information end

func conditional_switch_001_T(__taint_src string) {
	switch 2 {
	case 2:
		__taint_sink(__taint_src)
	}
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
	}

func main() {
    __taint_src := "taint_src_value"
    conditional_switch_001_T(__taint_src)
}

File: sast-php/case/completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_001_T.php
<?php
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用追踪->控制流
// scene introduction = switch-case 分支中污点传递
// level = 2
// bind_url = completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_001_T
// evaluation information end

function conditional_switch_001_T($__taint_src) {
    $result = "";
    $type = "exec";
    switch ($type) {
        case "exec":
            $result = $__taint_src;
            break;
        case "safe":
            $result = "safe_value";
            break;
    }
    __taint_sink($result);
}

function __taint_sink($o) {
    shell_exec($o);
}

$taint_src = "taint_src_value";
conditional_switch_001_T($taint_src);
